### DISCUSSION TITLE
Option to automatic reload config

### DISCUSSION THREAD

**Discussion by jvortmann** [ghostty-org/ghostty#3963] [Feature Requests, Ideas] [answered]
I believe a setting to automatic reload config would be nice to quickly set them up.
It would allow the config to be automatically reloaded on file save.

The options could encompass the following:
```
config-auto-reload: none | theme | all
```
With:
1. `none` - current behavior, only manual or keybinding for reload the config
2. `theme` - only the theme is reloaded if a file change is detected in the config file
3. `all` - all the settings are reloaded

3. option allows a quick setup and 2. allow a stable setup able to swap themes quickly

**Comment by JakeWharton** [comment, marked as answer]
See https://github.com/ghostty-org/ghostty/discussions/3643

**Reply by alienman5k** [reply]
It would be great to have a link to what was previous discussed to understand why the project does not want to support this feature, so we don't keep asking the same over and over.
I would still love that this is really considered as a feature for the future.

**Reply by sowmith1729** [reply]
Yeah, I couldn't find any discussion about where it was discussed before, #3643 just leads to a message saying it was discussed. The closest thing is `ghostty +list-keybinds | grep reload` and using that keybind.

**Reply by 00-kat** [reply]
I've seen it discussed before on the Discord, but I am very doubtful that I would be able to find it again given the supreme quality of Discord's search /s.
